### ISSUE TITLE
Latest release (0.18.0) points to wrong image (local/opentelemetry-operator:e2e)

### ISSUE THREAD

**Issue by gramidt** [open-telemetry/opentelemetry-operator#170]
The latest 0.18.0 release is pointing to the wrong image. Expectation is that it points to `quay.io/opentelemetry/opentelemetry-operator`. 

```
Events:
  Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
...
  Normal   Pulling    19s (x3 over 58s)  kubelet            Pulling image "local/opentelemetry-operator:e2e"
  Warning  Failed     18s (x3 over 57s)  kubelet            Failed to pull image "local/opentelemetry-operator:e2e": rpc error: code = Unknown desc = Error response from daemon: pull access denied for local/opentelemetry-operator, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
  Warning  Failed     18s (x3 over 57s)  kubelet            Error: ErrImagePull
  Normal   BackOff    6s (x3 over 56s)   kubelet            Back-off pulling image "local/opentelemetry-operator:e2e"
  Warning  Failed     6s (x3 over 56s)   kubelet            Error: ImagePullBackOff
```

**Comment by gramidt** [comment]
@jpkrohling - I'll try to dig in and determine what may have went wrong with the release process.

**Comment by jpkrohling** [comment]
This happens because the e2e step will override the image to use, here:
https://github.com/open-telemetry/opentelemetry-operator/blob/e23224ee92ac86d9b31d70dc580a25c1883d2d9d/Makefile#L105

The target `set-image-controller` will then change the kustomize file here:
https://github.com/open-telemetry/opentelemetry-operator/blob/e23224ee92ac86d9b31d70dc580a25c1883d2d9d/Makefile#L68

On previous versions, we weren't changing it as we didn't have e2e tests. But now, either the publish step should use a fresh checkout, or we have to revert the changes before publishing the artifacts.

IIRC, this is the file that would need to be reverted: https://github.com/open-telemetry/opentelemetry-operator/blob/e23224ee92ac86d9b31d70dc580a25c1883d2d9d/config/manager/kustomization.yaml#L2

**Comment by gramidt** [comment]
@jpkrohling - I apologize, but may be misunderstanding you. I believe the issue stems from `set-test-image-vars` target is incorrect. First `IMG` is not indented with a tab, thus causing `IMG` to always be set to "local/opentelemetry-operator:e2e" for the entire Makefile. Second variable assignments in targets do not affect other targets by default. To get this to work as is without additional restructuring, `eval` is needed to set `IMG` globally, so the updated value is accessible in `set-image-controller`. I opened #171 yesterday with the this proposed change for your review.

**Comment by jpkrohling** [comment]
Sounds good. Would be good to confirm the cause of the issue by simulating the release workflow locally. If you can confirm that, and confirm that the fix works, I'll merge the PR and release 0.18.1

**Comment by gramidt** [comment]
For future reference: https://github.com/open-telemetry/opentelemetry-operator/pull/171#discussion_r563197209
